conftest.py

```diff
@@ -149,92 +145,150 @@ def _send_post(
     return _send_post
 
 
-@pytest.fixture(scope = "session")
+@pytest.fixture(scope="session")
 def make_valid_request(send_post):
-    def _make_valid_request(
-        method: SensorMethod, params: dict | None = None
-    ) -> dict:
+    def _make_valid_request(method: SensorMethod, params: dict | None = None) -> dict:
         payload = make_valid_payload(method=method, params=params)
         sensor_response = send_post(**payload)
-        return sensor_response.get("result", {})
+        return sensor_response
 
     return _make_valid_request
 
 
-@pytest.fixture(scope = "session")
+@pytest.fixture(scope="session")
 def get_sensor_info(make_valid_request):
     def _get_sensor_info():
         log.info("Get sensor info")
         sensor_response = make_valid_request(SensorMethod.GET_INFO)
-        return SensorInfo(**sensor_response)
-    
+        result = None
+
+        if "result" in sensor_response and isinstance(sensor_response["result"], dict):
+            result = SensorInfo(**sensor_response["result"])
+
+        if "error" in sensor_response:
+            result = sensor_response["error"]
+
+        return result
+
     return _get_sensor_info
 
 
 @pytest.fixture(scope="session")
 def get_sensor_reading(make_valid_request):
     def _get_sensor_reading():
         log.info("Get sensor reading")
-        return make_valid_request(SensorMethod.GET_READING)
+        sensor_response = make_valid_request(SensorMethod.GET_READING)
+        result = None
+
+        if isinstance(sensor_response, dict):
+            if "result" in sensor_response:
+                if isinstance(sensor_response["result"], float):
+                    result = sensor_response["result"]
+
+        if "error" in sensor_response:
+            result = sensor_response["error"]
+
+        return result
 
     return _get_sensor_reading
 
 
-@pytest.fixture
+@pytest.fixture(scope="session")
 def set_sensor_name(make_valid_request):
-    def _get_sensor_name(name: str):
+    def _set_sensor_name(name: str):
         log.info("Set sensor name to %s", name)
-        return make_valid_request(SensorMethod.SET_NAME, {"name": name})
+        sensor_response = make_valid_request(SensorMethod.SET_NAME, {"name": name})
+        result = None
 
-    return _get_sensor_name
+        if "result" in sensor_response:
+            result = SensorInfo(**sensor_response["result"])
+
+        if "error" in sensor_response:
+            result = sensor_response["error"]
+
+        return result
+
+    return _set_sensor_name
 
 
 @pytest.fixture(scope="session")
 def get_sensor_methods(make_valid_request):
     def _get_sensor_methods():
         log.info("Get sensor methods")
-        return make_valid_request(SensorMethod.GET_METHODS)
+        sensor_response = make_valid_request(SensorMethod.GET_METHODS)
+        result = None
+
+        if "result" in sensor_response:
+            result = SensorInfo(**sensor_response["result"])
```

Comment:
`get_sensor_methods` doesn't return sensor info, so we shouldn't try to instantiate a `SensorInfo` object here. Since we don't have a class definition for what this sensor method returns, we just need to return `sensor_response["result"]`.